tee_supplicant/src/tee_supplicant.c

```diff
@@ -342,8 +351,23 @@ static int tee_supp_init(const struct device *dev)
 
 	sys_dlist_init(&shm_list);
 
+	if (tee_get_version(tee_dev, &info)) {
+		LOG_ERR("Unable to allocate thread argument");
+		return -EINVAL;
+	}
+	arg = k_malloc(sizeof(struct thread_arg));
```

Comment:
If you check caps only on init - then you don't need arg anymore. Also please remove arg from the parameters list all the way down to shm_alloc as nonused variable.

Reply:
I have removed arg from parameter list for shm_alloc. Actually arg in this PR isn't used but I didn't remove at all because it is used in next patch set

Reply:
Then this should be added in the following patchset. Also, I're calling process_request(dev, arg) (line 333) when process_request accepts only dev parameter. This is not going to compile.

Reply:
`thread_arg` structure moved to the next patch set